optaplanner-core/src/main/java/org/optaplanner/core/impl/constructionheuristic/DefaultConstructionHeuristicPhase.java

```diff
@@ -36,6 +36,8 @@
 public class DefaultConstructionHeuristicPhase<Solution_> extends AbstractPhase<Solution_>
         implements ConstructionHeuristicPhase<Solution_> {
 
+    private final BestSolutionRecaller<Solution_> bestSolutionRecaller;
```

Comment:
Some phases still need the recaller, because they call their own special methods on it.

Reply:
For my understanding: which phases call which special methods on it?

Reply:
I guess the CH because it doesn't do best solution events for every step?